src/redux.js

```diff
@@ -114,24 +106,15 @@ function* createSession(action) {
     }
 }
 
-function* signupSlack(action) {
-    try {
-        yield call(waitFor, getSessionCreatedFromState, true)
-        const req = request.post(`${API_URL}/slack/invite`)
-                .send({ email: action.payload.email })
-        const response = yield req
-        yield put({ type: actionTypes.SIGNUP_SLACK_SUCCESS, response })
-    } catch (error) {
-        yield put({ type: actionTypes.SIGNUP_SLACK_FAILURE, error })
-        console.error(error)
-    }
+function* signupDiscord(action) {
+    yield put(actions.signupMailingList({ discord_signup: true, ...action.payload }))
 }
 
 function* signupMailingList(action) {
     try {
         yield call(waitFor, getSessionCreatedFromState, true)
         const req = request.post(`${API_URL}/mailing-list/signup`)
-                .send({ email: action.payload.email })
+                .send(action.payload)
```

Comment:
I'll say, I don't love this design because it pushes the concern of “what does the signup endpoint take” way into whoever originally triggers the action. I would expect the action to have its interface (probably in camelCase, so it would take `email` and `discordSignup` for its payload), while `signupMailingList` would be charged with taking that payload and filtering/adapting to the particular bits the API endpoint expects.

That said, probably not worth blocking on it.

Reply:
Good points! I actually thought similarly about obscuring the expected request body of the mailing signup endpoint but did it anyway (sorry!). 🙈😬 I also thought about dropping the dedicated action for `signupDiscord` and just using the `signupMailingList` one with the `discord_signup` param, but then it would push that logic to the view and I didn't really want to do that. Also, I'll keep in mind the casing for the future. I go back and forth about whether all JS variables should be camelCase or whether it it's easier to understand if they mirror the expected payload.

I'm going to be revisiting redux soon in my refactor to Gatsby (not sure if it makes sense to keep it yet in that ecosystem) - if it ends up staying I will polish this up.